src/System.Private.CoreLib/shared/System/Text/Unicode/Utf8.cs

```diff
@@ -0,0 +1,186 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.Unicode
+{
+    public static class Utf8
```

Comment:
This is adding new public APIs. Do we have an approved API review issue for these?

Reply:
The `Utf8` static class is approved. See https://github.com/dotnet/corefx/issues/34761 and https://github.com/dotnet/apireviews/pull/89.

Reply:
I would be better to submit that as separate PR from the experimental Utf8 string with appropriate tests, etc.

Reply:
Opened https://github.com/dotnet/coreclr/pull/23219 to track this.